pkg/proxy/proxy.go

```diff
@@ -89,25 +90,23 @@ func (p *Proxy) get(name, namespace string) (*release.Release, error) {
 		helm.ReleaseListFilter(name),
 		helm.ReleaseListNamespace(namespace),
 		helm.ReleaseListStatuses(allReleaseStatuses),
+		// Get just the latest release
+		helm.ReleaseListLimit(1),
+		helm.ReleaseListSort(int32(services.ListSort_LAST_RELEASED)),
+		helm.ReleaseListOrder(int32(services.ListSort_DESC)),
 	)
 	if err != nil {
 		return nil, fmt.Errorf("Unable to list helm releases: %v", err)
 	}
-	var rel *release.Release
 	if list != nil {
 		if l := list.GetReleases(); l != nil {
-			for _, r := range l {
-				if (namespace == "" || namespace == r.Namespace) && r.Name == name {
-					rel = r
-					break
-				}
+			r := l[0]
+			if (namespace == "" || namespace == r.Namespace) && r.Name == name {
```

Comment:
Why do we need to check for the namespace and name again here, when we already are filtering it in ListReleases?

Reply:
the problem is that some unit tests rely in that logic (to double filter the results) but that should not be necessary.